wasm/naivevm/vm.go

```diff
@@ -112,9 +112,17 @@ func (vm *naiveVirtualMachine) Compile(f *wasm.FunctionInstance) error {
 			vm.popFrame()
 			return returns, nil
 		}
-	} else {
-		// Compile native Wasm functions.
-		f.Body[len(f.Body)-1] = byte(wasm.OptCodeReturn)
+	} else { // Compile native Wasm functions.
+
+		lastIndex := len(f.Body) - 1
+		switch f.Body[lastIndex] {
+		case wasm.OpcodeReturn:
+		case wasm.OpcodeEnd: // coerce end to return: TODO why?
```

Comment:
This is the new check. all other change is mechanical.

Reply:
do we need to have this swtich? From the spec, I understand that every function end with "end" opcode so we shouldn't reach case `wasm.OpcodeReturn` above. Maybe I am missing something though

Reply:
ok I will check to see if anything hits "return" if not, then take out the case statement!

Reply:
at least our tests hit the return branch... if I panic, one test dies

```
--- FAIL: TestSpecification (0.57s)
    --- FAIL: TestSpecification/imports.wast (0.00s)
        --- FAIL: TestSpecification/imports.wast/naivevm (0.00s)
            --- FAIL: TestSpecification/imports.wast/naivevm/module/line:24 (0.00s)
panic: return at end [recovered]
	panic: return at end

goroutine 30428 [running]:
```

so, there are a couple questions I guess.
* is it a bug in translation that ^^ ends with a return?
* for the cases that end in .. "end", why do we change that to return? (I need to add a comment per TODO)

Reply:
> is it a bug in translation that ^^ ends with a return?
naivevm doesn't translate so I think my understanding of spec was wrong. Function can end with "return" instead of "end"

> for the cases that end in .. "end", why do we change that to return? (I need to add a comment per TODO)
 
that's just because I implemented the naivevm in the way that all the function must end with "return" :D

Reply:
will revert this per https://github.com/tetratelabs/wazero/issues/79 as we don't need to spend time on dead code